geomesa-jupyter/src/main/scala/org/locationtech/geomesa/jupyter/Leaflet.scala

```diff
@@ -0,0 +1,174 @@
+    override def render: String = {
+      val coords = poly.getCoordinates.map { c => Array(c.y, c.x).mkString("[", ",", "]") }.mkString("[", ",", "]")
+      s"L.polygon($coords, { fillOpacity: 0 }).addTo(map);"
+    }
+  }
+
+  implicit class JTSPointLayer(val point: Point) extends GeoRenderable {
+    override def render: String = s"L.circle([${point.getY},${point.getX}], 5, {}).addTo(map);"
+  }
+
+  implicit class JTSLineStringLayer(val ls: LineString) extends GeoRenderable {
+    override def render: String = {
+      val coords = ls.getCoordinates.map { c => Array(c.y, c.x).mkString("[", ",", "]") }.mkString("[", ",", "]")
+      s"L.polyline($coords).addTo(map);"
+    }
+  }
+
+  // TODO: parameterize base url for js and css
+  def buildMap(layers: Seq[GeoRenderable], center: (Double, Double) = (0,0), zoom: Int = 8) =
+    s"""
+       |<html>
+       |  <head>
+       |    <link rel="stylesheet" href="js/leaflet.css" />
+       |    <script src="js/leaflet.js"></script>
+       |    <script src="js/leaflet.wms.js"></script>
+       |    <script src="js/countries.geo.json" type="text/javascript"></script>
+       |  </head>
+       |  <body>
+       |    <div id='map' style="width:100%;height:500px"></div>
+       |    <script>
+       |      // Initialize the Base Layer... Loaded from GeoJson
+       |      var basestyle = {"color": "#717171", "weight": 2, "opacity": 1.0};
+       |      var base = L.geoJson(worldMap, basestyle);
+       |
+       |      //'map' is the id of the map
+       |      var map = L.map('map', {
+       |        crs: L.CRS.EPSG4326,
+       |        center: [${center._1}, ${center._2}],
+       |        zoom: ${zoom}
+       |      });
+       |
+       |      map.addLayer(base);
+       |      ${layers.map(_.render).mkString(sep = "\n")}
+       |
+       |    </script>
+       |  </body>
+       |</html>
+     """.stripMargin
+
+  def render(layers: Seq[GeoRenderable], center: (Double, Double) = (0,0), zoom: Int = 8) = {
+    val id = org.apache.commons.lang3.RandomStringUtils.randomAlphanumeric(5)
+    s"""
+       |<iframe id="${id}" sandbox="allow-scripts allow-same-origin" style="border:none;width=100%;height:520px" srcdoc="${xml.Utility.escape(buildMap(layers, center, zoom))}"></iframe>
+       |<script>
+       |  if(typeof resizeIFrame != 'function') {
+       |    function resizeIFrame(el, k) {
+       |      el.style.height = el.contentWindow.document.body.scrollHeight + 'px';
+       |      if(k<=3) { setTimeout(function() { resizeIFrame(el, k+1)}, 1000) };
+       |    }
+       |  }
+       |  $$().ready(function() {resizeIFrame($$('#$id').get(0).1);});
+       |</script>
+     """.stripMargin
+  }
+
+  def show(layers: Seq[GeoRenderable], center: (Double, Double) = (0,0), zoom: Int = 8)(implicit disp: String => Unit) = disp(render(layers,center,zoom))
+
+  def print(layers: Seq[GeoRenderable], center: (Double, Double) = (0,0), zoom: Int = 8) = println(buildMap(layers,center,zoom))
+}
```

Comment:
well, if you ever want to touch a file w/ POSIX compliant tooling it should have a newline.